src/fsharp/PrettyNaming.fs

```diff
@@ -109,66 +109,162 @@ module internal Microsoft.FSharp.Compiler.PrettyNaming
         ( ')', "RParen");
         ( ' ', "Space");
         ( '[', "LBrack");
-        ( ']', "RBrack"); ]
+        ( ']', "RBrack"); |]
 
-    let private opCharDict = 
-        let t = new Dictionary<_,_>()
-        for (c,_) in opCharTranslateTable do 
-            t.Add(c,1)
+    /// The set of characters usable in custom operators.
+    let private opCharSet =
+        let t = new HashSet<_>()
+        for (c,_) in opCharTranslateTable do
+            t.Add(c) |> ignore
         t
         
-    let IsOpName (n:string) =
-        let rec loop i = (i < n.Length && (opCharDict.ContainsKey(n.[i]) || loop (i+1)))
+    let IsOpName (name:string) =
+        let nameLen = name.Length
+        let rec loop i = (i < nameLen && (opCharSet.Contains(name.[i]) || loop (i+1)))
```

Comment:
I can't recall if this is perf critical, but since it might well be I think it's better to leave it as it was